### PR TITLE
[Storage][ADLS] API updates

### PR DESCRIPTION
**Pull request by annatisch**
Some tweaks based on feedback from API review: https://github.com/Azure/azure-sdk-for-python/pull/9987

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    :ivar int size:
+        The size of the total data in the stream. This will be the byte range if speficied,
+        otherwise the total size of the file.
+    """
+
+    def __init__(self, downloader):
+        self._downloader = downloader
+        self.name = self._downloader.name
+        self.properties = FileProperties._from_blob_properties(self._downloader.properties) 
+        self.size = self._downloader.size
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py]: probably we should add # pylint: disable=protected-access

**Comment by xiafu-msft** [comment]
Thanks so much for helping add this wrapper! It looks great!

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py [hunk 2]
```diff
@@ -545,8 +542,8 @@ def read_file(self, offset=None,   # type: Optional[int]
             The timeout parameter is expressed in seconds. This method may make
             multiple calls to the Azure service and the timeout will apply to
             each call individually.
-        :returns: downloaded data or the size of data written into the provided stream
-        :rtype: bytes or int
+        :returns: A streaming object (StorageStreamDownloader)
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py]: Do you think we should add anything about readall here?

Comment (rakshith91) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py]: I mean in the docstrings for download

Comment (annatisch) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py]: It's mentioned at the top of the docstring

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from ._models import FileProperties
+
+
+class StorageStreamDownloader(object):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py]: Should we move this into the _shared folder?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py [hunk 4]
```diff
@@ -502,22 +503,18 @@ def flush_data(self, offset,  # type: int
         except StorageErrorException as error:
             process_storage_error(error)
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py]: Should we also update any samples @xiafu-msft added recently?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py (hunk 3 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py]: No, as it's a custom ADLS-specific implementation that's dependent on the Blobs-specific implementation.
Though potentially we can refactor a common base to add to the _shared directory later.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py (hunk 4 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py]: @xiafu-msft I think the upload/download sample I updated is the only one?

**Pull request merged**